processor/src/main/java/com/linecorp/decaton/processor/metrics/Metrics.java

```diff
@@ -79,14 +76,21 @@ protected AbstractMetrics() {
         @Override
         public void close() {
             synchronized (meterRefCounts) {
-                for (Meter meter : meters) {
-                    Meter.Id id = meter.getId();
-                    int count = meterRefCounts.get(id).decrementAndGet();
-                    if (count == 0) {
+                // traverse from the end to avoid arrayCopy
+                for (ListIterator<Meter> iterator = meters.listIterator(meters.size() - 1); iterator.hasNext(); ) {
```

Comment:
`iterator.hasPrevious`?

Reply:
and starting index should be `meters.size()`?

<img width="761" alt="image" src="https://user-images.githubusercontent.com/2255161/89015812-abb89080-d352-11ea-9283-125dd60f4be7.png">

Reply:
yes to both..